compiler+runtime/src/cpp/jank/aot.cpp

```diff
@@ -0,0 +1,281 @@
+    jank_load_clojure_string_native();
+    jank_load_jank_compiler_native();
+
+    )");
+
+    for(auto const &m : __rt_ctx->loaded_modules_in_order)
+    {
+      sb(module::module_to_load_function(m));
+      sb(R"(();
+      )");
+    }
+
+    sb(R"(
+
+    // project modules loaded
+
+    // main function being called
+    
+    auto const fn(jank_var_intern_c(")");
+
+    sb(module);
+    sb(R"(", "-main"));
+    jank_call0(jank_deref(fn));
+
+  } };
+
+  return jank::init(argc, argv, /* init_default_ctx= */ true, fn);
+}
+  )");
+
+    auto const main_file(relative_to_cache_dir("/jank_main.cpp"));
+
+    std::ofstream out(main_file);
+    out << sb.release();
+
+    out.flush();
+    out.close();
+  }
+
+  void aot_compiler::compile(jtl::immutable_string const &module) const
+  {
+    gen_entrypoint(module);
+
+    llvm::IntrusiveRefCntPtr<clang::DiagnosticOptions> DiagOpts = new clang::DiagnosticOptions();
+
+    auto *DiagClient = new clang::TextDiagnosticPrinter(llvm::errs(), &*DiagOpts);
+
+    llvm::IntrusiveRefCntPtr<clang::DiagnosticIDs> DiagID(new clang::DiagnosticIDs());
+
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS = llvm::vfs::getRealFileSystem();
+
+    clang::DiagnosticsEngine Diags(DiagID, &*DiagOpts, DiagClient, /*ShouldOwnClient=*/true);
+
+
+    std::string const TargetTripleStr = llvm::sys::getDefaultTargetTriple();
+    llvm::errs() << "Using target triple: " << TargetTripleStr << "\n";
+
+    /* The Driver needs the path to the executable (used for finding related tools/resources)
+     * For this example, we'll use argv[0] or a placeholder. This might need adjustment
+     * depending on how/where your program is run relative to clang resources.
+     * TODO: infer clang++ path. Will be OS dependent and will require to infer correct version's path */
+    auto const ClangExecutablePath{ "/usr/bin/clang++" };
+    clang::driver::Driver TheDriver(ClangExecutablePath,
+                                    TargetTripleStr,
+                                    Diags,
+                                    "jank_aot_compilation",
+                                    VFS);
+    TheDriver.setCheckInputsExist(false);
+
+    std::vector<char const *> Args = { strdup(ClangExecutablePath) };
+
+    for(auto const &module : __rt_ctx->loaded_modules_in_order)
+    {
+      auto const &module_path{ module::module_to_path(module) };
+      util::string_builder sb;
+
+      Args.push_back(strdup(relative_to_cache_dir(module_path).c_str()));
+    }
+
+    Args.push_back(strdup(relative_to_cache_dir("jank_main.cpp").c_str()));
+
+    for(auto const &include_dir : include_dirs)
+    {
+      Args.push_back(strdup(util::format("-I{}", include_dir).c_str()));
+    }
+
+    for(auto const &library_dir : library_dirs)
+    {
+      Args.push_back(strdup(util::format("-L{}", library_dir).c_str()));
+    }
+
+    for(auto const &lib : { "-ljank",
+                            // Default libs that Jank depends on
```

Comment:
jank is always lowercase. Also, please use `/* */` comments and use correct punctuation. Comments are documentation.